File: frontend/src/types/question.ts
export type Difficulty = 'Easy' | 'Medium' | 'Hard'

export type CorrectOption = 'A' | 'B' | 'C' | 'D'

export interface Question {
  id: number
  text: string
  optionA: string
  optionB: string
  optionC: string
  optionD: string
  correctOption: CorrectOption
  difficulty: Difficulty
  category: string
  explanation: string | null
  createdAt: string
  /** null only for legacy questions not yet assigned to a book */
  bookId: number | null
  bookTitle: string | null
  /** Optional local illustration (/question-images/...). Both null or both set. */
  imageUrl: string | null
  imageAltText: string | null
}

/** Write shape: every new or updated question must name its book. */
export type QuestionInput = Omit<Question, 'id' | 'createdAt' | 'bookId' | 'bookTitle'> & { bookId: number }

File: backend/LiteratureMillionaire.API/Entities/Question.cs
namespace LiteratureMillionaire.API.Entities;

public class Question
{
    public int Id { get; set; }
    public string Text { get; set; } = string.Empty;
    public string OptionA { get; set; } = string.Empty;
    public string OptionB { get; set; } = string.Empty;
    public string OptionC { get; set; } = string.Empty;
    public string OptionD { get; set; } = string.Empty;

    /// <summary>Single character: A, B, C or D.</summary>
    public char CorrectOption { get; set; }

    public Difficulty Difficulty { get; set; }
    public string Category { get; set; } = string.Empty;
    public string? Explanation { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    /// <summary>
    /// Owning book. Nullable only during the expand-contract transition: legacy
    /// questions created before books existed stay unassigned until classified.
    /// New and updated questions must always carry a BookId (enforced in the API).
    /// </summary>
    public int? BookId { get; set; }
    public Book? Book { get; set; }

    /// <summary>
    /// Quiz mode the question is played in ("bilik-dunyasi", "ayin-kitabi", ...). Null for legacy questions, which
    /// are never selected for a game. Not the same as <see cref="Category"/>, the source sub-category.
    /// </summary>
    public int? QuizModeId { get; set; }
    public QuizMode? QuizMode { get; set; }

    /// <summary>
    /// Optional illustration, a local frontend asset such as "/question-images/name.webp".
    /// Either both media fields are set or both are null (enforced in the API and by a CHECK constraint).
    /// </summary>
    public string? ImageUrl { get; set; }
    public string? ImageAltText { get; set; }
}
